sw/device/lib/dif/dif_edn.c

```diff
@@ -267,45 +275,51 @@ dif_result_t dif_edn_get_main_state_machine(const dif_edn_t *edn,
 
 dif_result_t dif_edn_instantiate(
     const dif_edn_t *edn, dif_edn_entropy_src_toggle_t entropy_src_enable,
-    const dif_edn_seed_material_t *seed_material) {
-  if (edn == NULL) {
+    const dif_edn_cmd_t *cmd) {
+  if (edn == NULL || cmd == NULL) {
     return kDifBadArg;
   }
+  dif_csrng_seed_material_t csrng_seed_material;
+  csrng_seed_material.seed_material_len = cmd->len;
+  memcpy(&csrng_seed_material.seed_material, cmd->data,
+         sizeof(csrng_seed_material.seed_material));
   return csrng_send_app_cmd(
       edn->base_addr, EDN_SW_CMD_REQ_REG_OFFSET,
       (csrng_app_cmd_t){
           .id = kCsrngAppCmdInstantiate,
           .entropy_src_enable =
               (dif_csrng_entropy_src_toggle_t)entropy_src_enable,
-          .seed_material = (const dif_csrng_seed_material_t *)seed_material,
+          .seed_material = &csrng_seed_material,
       });
 }
 
-dif_result_t dif_edn_reseed(const dif_edn_t *edn,
```

Comment:
`dif_edn_reseed()` and `dif_edn_update()` should be able to continue using `dif_edn_seed_material_t` to make the interface more intuitive. I agree `dif_edn_auto_params_t` should use `dif_edn_cmd_t` instead.